### PR TITLE
Fix camper edit functionality

### PR DESCRIPTION
**Pull request by twilkhoo**
## Notion ticket link

[Fix camper edit functionality](https://www.notion.so/uwblueprintexecs/Fix-camper-edit-functionality-12b9fad2bae24cce9cf7352535ebc406)

## Implementation description
* Added a useEffect to ensure the changes were loaded every time there was a new camper clicked
* Added a useMemo to keep track of initial state to set back to if an admin clicks cancel (very similar to what I did for all the edit components in the ReviewRegistration step, registrant experience).

## Steps to test
1. Navigate to any camp with campers, http://localhost:3000/admin/camp/6397f94e2ff11dcda0ba4960 works well.
2. Try to edit (multiple) campers, the initial information loaded should all be correct now and of the clicked camper.
3. Click save, cancel, click outside of the modal (another way to cancel), and ensure everything behaves as expected now.

## What should reviewers focus on?
* Correctness, functionality.

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: backend/typescript/middlewares/validators/adminValidators.ts [hunk 1]
```diff
@@ -2,6 +2,7 @@ import { Request, Response, NextFunction } from "express";
 import { validateFormQuestion } from "./formQuestionValidators";
 import { validatePrimitive, getApiValidationError } from "./util";
 
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (twilkhoo) [inline backend/typescript/middlewares/validators/adminValidators.ts:5]: Disabled whatever eslint `any` checks that have been in the codebase forever, thought it was about time we do so.

File: frontend/src/components/pages/CampOverview/EditCamperModal/index.tsx [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import React, { useState } from "react";
+import React, { useEffect, useMemo, useState } from "react";
```
Comment (twilkhoo) [inline frontend/src/components/pages/CampOverview/EditCamperModal/index.tsx:1]: All the code regarding the actual edit modal is in this file.

File: backend/typescript/middlewares/validators/adminValidators.ts (hunk 1 above)
Comment (rchang0501) [reply backend/typescript/middlewares/validators/adminValidators.ts:5]: sg!

**Comment by rchang0501** [comment]
I found a bug - upon hitting edit, sometimes the modal that comes up is for the first camper's initial state. 

In this case, I am trying to select the last camper "test test" - when I hit edit the modal brings up the state for "lkahf oehba" which was the initial state of the first camper (the one with the name "Rachel Wormland") 

Steps to recreate: 
1. select a camper and hit edit (it will be fine)
2. leave the modal
3. select the same camper again and hit edit (the modal will now show the initial state of the first camper) 


https://user-images.githubusercontent.com/46568041/217083444-3fa30fce-1562-43c1-b293-108725b430d1.mov

**Comment by rchang0501** [comment]
> I found a bug - upon hitting edit a second time in a row, the modal that comes up is for the first camper's initial state.
> 
> In this case, I am trying to select the last camper "test test" - when I hit edit the modal brings up the state for "lkahf oehba" which was the initial state of the first camper (the one with the name "Rachel Wormland")
> 
> Steps to recreate: (done on http://localhost:3000/admin/camp/63cb52d3b74ed1b551094ca7)
> 
> 1. select a camper and hit edit (it will be fine)
> 2. leave the modal
> 3. select the same camper again and hit edit (the modal will now show the initial state of the first camper)
> 
>  Screen.Recording.2023-02-06.at.3.52.40.PM.mov

Played with it some more - upon refresh of the page the same behaviour is exhibited but with the updated information for the first camper. 

https://user-images.githubusercontent.com/46568041/217087397-75706dd5-8f80-435a-b0e7-ece1c309c770.mov

File: frontend/src/components/pages/CampOverview/EditCamperModal/index.tsx [hunk 3]
```diff
@@ -69,6 +69,9 @@ const EditCamperModal = ({
     formResponses,
   };
 
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const initialStateVariables = useMemo(() => stateVariables, []);
```
Comment (rchang0501) [inline frontend/src/components/pages/CampOverview/EditCamperModal/index.tsx:73]: the initial state needs to be updated everytime we select a new camper or the afformentioned bug will occurr. I solved it by passing in all the fields as dependencies to the useMemo 
```
  const initialStateVariables = useMemo(() => stateVariables, [
    firstName,
    lastName,
    age,
    allergies,
    specialNeeds,
    formResponses,
  ]);
  ```

Comment (rchang0501) [reply frontend/src/components/pages/CampOverview/EditCamperModal/index.tsx:73]: Basically, the issue is that the initial state doesn't get updated properly between renders. Whichever camper I selected first to edit, that camper's initial fields will be saved in `initialStateVariables` so to make sure it's updated between renders I just passed all the selected camper's fields as dependencies. 

We don't need to use this exact implementation but the idea is we need to ensure `initialStateVariables` between renders

**Review by rchang0501 — CHANGES_REQUESTED** [review]
Just need to fix the `initialStateVariables` bug and we should be good!

**Comment by twilkhoo** [comment]
Alright I took care of the problem, after testing pretty extensively this time I decided to use a updateMemo just like we do in the EditCards on the review page.

https://user-images.githubusercontent.com/30396273/217136616-fff4e5f6-3754-426e-a2fc-11e525703912.mp4

**Review by rchang0501 — APPROVED** [review]
tested, everything looks great!

**Pull request merged**